prompt_toolkit/formatted_text/ansi.py

```diff
@@ -257,11 +258,17 @@ def format(self, *args: str, **kwargs: str) -> "ANSI":
         Like `str.format`, but make sure that the arguments are properly
         escaped. (No ANSI escapes can be injected.)
         """
-        # Escape all the arguments.
-        args = tuple(ansi_escape(a) for a in args)
-        kwargs = {k: ansi_escape(v) for k, v in kwargs.items()}
+        return ANSI(FORMATTER.vformat(self.value, args, kwargs))
 
-        return ANSI(self.value.format(*args, **kwargs))
+    def __mod__(self, value: Union[object, Tuple[object, ...]]) -> "ANSI":
```

Comment:
The type for ``value`` can be ``object`` (A union with an ``object`` inside is always ``object``.)